src/org/exist/xmlrpc/SerializedResult.java

```diff
@@ -19,26 +19,28 @@
  */
 package org.exist.xmlrpc;
 
-import org.exist.util.VirtualTempFile;
+import org.exist.util.io.TemporaryFileManager;
 import org.exist.xquery.XPathException;
 
+import java.nio.file.Path;
+
 /**
  * Simple container for the results of a query. Used to cache
  * query results that may be retrieved later by the client.
  *
  * @author jmfernandez
  */
 public class SerializedResult extends AbstractCachedResult {
-    protected VirtualTempFile result;
+    protected Path result;
```

Comment:
so here we have a 'real' difference: VirtualTempFile could hold a complete document in memory as where Path always will have a resource on disk. @adamretter you are convinced that typically the OS cache will handle this transparently right?  Only when a virus scanner is involved, having a file might have additional performance impact no?

Reply:
Well actually `VirtualTempFile` could have both an on-disk temp-file and an in-memory copy.

The memory use is now controllable because we are using `CachingFilterInputStream` and you can choose the cache implementation in `conf.xml`.